addons/website/models/ir_qweb.py

```diff
@@ -110,6 +114,54 @@ def _post_processing_att(self, tagName, atts):
         if not website:
             return atts
 
+        if (
+            website.cookies_bar
+            and website.block_third_party_domains
+            and not self.env.context.get('cookies_allowed')
+        ):
+            # If the cookie banner is activated, 3rd-party embedded iframes and
+            # scripts should be controlled. As such:
+            # - 'domains' is a watchlist on the iframe/script's src itself,
+            # - 'custom_domains' is a watchlist set by the user on the iframe /
+            # script's src,
+            # - 'classes' is a watchlist on container elements in which iframes
+            # are/could be built on the fly client-side for some reason.
+            cookiesWatchList = {
+                'domains': website.default_blocked_third_party_domains.split('\n'),
+                'custom_domains': website.sudo().blocked_third_party_domains.split("\n"),
+                'classes': {
+                    's_map',
+                    's_instagram_page',
+                    'o_facebook_page',
+                    'o_background_video',
+                    'media_iframe_video',
+                },
+            }
+            remove_src = False
+            if tagName in ('iframe', 'script'):
+                srcHost = parse_url((atts.get('src') or '').lower()).host
+                if srcHost:
+                    all_domains = (
+                        [re.escape(domain) for domain in cookiesWatchList['domains']]
+                        + [re.escape(domain) for domain in cookiesWatchList['custom_domains']]
+                    )
+                    # Match domains with or without subdomain, or just second-
+                    # level domains.
+                    domain_regex = re.compile(r'^(.*\.)?({})(\..*)?$'.format('|'.join(all_domains)))
+                    remove_src = bool(domain_regex.search(srcHost))
```

Comment:
> What is the (\..*)? part in the regex about?

Making sure I can match any TLD if none is given in the domains list. It's only there for Google domains, but it's necessary because you can technically have (legit) URLs for Google Maps with nearly any existing TLD (I say Maps as an example, but I guess it's the case for other Google services too).

> you tell users that you accept domains being written with their subdomain (www) but your regex seems to be wrong in that case no? You would still accept a https://example.com/hello if the user entered www.example.com as blocked domain no?

I didn't think about redirections from `www` subdomains to implicit subdomain or the other way around. But indeed I suppose that should be taken into account.

Reply:
> Making sure I can match any TLD if none is given in the domains list.

Oh, I see. Then I guess my `endswith` solution is not good enough. Although if we really want to be complete then writing "google" as a blocked "domain" actually blocks `google.mywebsite.be`, which is a bit sad :shrug: But probably overthinking, at worst cookie acceptance will be required for no reason. Just annoying that python does not seem to have a nice util to extract the main domain part and/or remove the TLD.

Probably anything is ok, I would just at least make sure the `www` is striped and add some precision in the comment (I missed the fact that the part about "second-level domain" is what you are explaining here).

Reply:
> "google" as a blocked "domain" actually blocks google.mywebsite.be, which is a bit sad

I thought about writing a slightly more complex regex that checks that the last bit is a true TLD (i.e. something like either `. + some letters + $`, or `. + 2-3 letters + . + 2 letters + $`). I ditched it because I didn't think it was necessary, but I can add it back if we want to avoid that problem.